src/pybind/mgr/volumes/fs/operations/versions/subvolume_v1.py

```diff
@@ -831,15 +831,25 @@ def snapshot_info(self, snapname):
                                   "snapshot name '{0}' is invalid".format(snapname))
         snappath = self.snapshot_data_path(snapname)
         snap_info = {}
+        nsize = 0
         try:
             snap_attrs = {'created_at':'ceph.snap.btime',
                           'data_pool':'ceph.dir.layout.pool'}
             for key, val in snap_attrs.items():
                 snap_info[key] = self.fs.getxattr(snappath, val)
+
             pending_clones_info = self.get_pending_clones(snapname)
             info_dict = {'created_at': str(datetime.fromtimestamp(float(snap_info['created_at']))),
                     'data_pool': snap_info['data_pool'].decode('utf-8')}
-            info_dict.update(pending_clones_info);
+            if self.state != SubvolumeStates.STATE_RETAINED:
```

Comment:
A comment or explanation in the commit message would be useful on why the size of snapshot isn't displayed in this case.